### PR TITLE
Feature/23 tags

### PR DESCRIPTION
**Pull request by mabiede**
- implement tags and tagged (mapping between entities and tags)
- add tags to contact detail overview
- only show the tag assignment form when allowed to do so

### CODE REVIEW

File: pool/app/tags/entity.ml [hunk 1]
```diff
@@ -1,29 +1,65 @@
...
+      (Description.create %> map CCOption.return)
+      description
+  in
+  let* title = Title.create title in
+  Ok { id; title; description }
+;;
+
+module Tagged = struct
+  type t =
+    { model : Model.t
```
Comment (timohuber) [inline pool/app/tags/entity.ml:58]: That information should be stored in the Tags.t type to differentiate the tags available for different models. 

As we use UUIDs, we do not need this information here.

File: pool/app/tags/tags.mli [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+val handle_event : Pool_database.Label.t -> event -> unit Lwt.t
+
+val find
+  :  Pool_database.Label.t
+  -> Id.t
+  -> (t, Pool_common.Message.error) result Lwt.t
+
+val find_all : Pool_database.Label.t -> t list Lwt.t
+
+val find_all_validated
```
Comment (timohuber) [inline pool/app/tags/tags.mli:79]: Not sure if we need to separate access for each tag. Maybe simplify and just create one right to assign / remove.

**Review by timohuber — COMMENTED** [review]
According to the issue, it should be possible to filter participants by tags. Are you planning to add that in a separate PR?

File: pool/app/tags/entity.ml (hunk 1 above)
Comment (mabiede) [reply pool/app/tags/entity.ml]: moved model information to tag itself

File: pool/cqrs_command/tags_command.ml [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+end = struct
+  type t = decoded
+
+  let handle ?(tags = Logs.Tag.empty) tag (command : t) =
+    let open Tags in
+    let tag : t =
+      { tag with
+        title = command.title
+      ; description = command.description
+      ; model = command.model
```
Comment (timohuber) [inline pool/cqrs_command/tags_command.ml:80]: Not sure if we should allow to change the model after the creation of the tags?

File: pool/web/view/page/page_admin_settings_tags.ml [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+  form
+    ~a:
+      [ a_method `Post
+      ; a_action (Sihl.Web.externalize_path action)
+      ; a_class [ "stack" ]
+      ; a_user_data "detect-unsaved-changes" ""
+      ]
+    [ Input.csrf_element csrf ()
+    ; div
+        ~a:[ a_class [ "stack-md" ] ]
```
Comment (timohuber) [inline pool/web/view/page/page_admin_settings_tags.ml:81]:
```suggestion
        ~a:[ a_class [ "grid-col-2" ] ]
```

**Review by timohuber — APPROVED** [review]

**Pull request merged**